src/OpenTK/Platform/MacOS/Cocoa/NSFloat.cs

```diff
@@ -179,5 +191,17 @@ struct NSRect
             return new RectangleF(s.Location, s.Size);
         } 
     }
-}
 
+    // Using IntPtr in NSFloat cause that if imported function
+    // return struct that consist of them you will get wrong data 
+    // This types are used for such function.
```

Comment:
If these are used as interop types, they'll need to be marked with [StructLayout(LayoutKind.Sequential)], otherwise the compiler is free to re-order in whatever way it wants.

If they're not used for interop, just use a Vector2/Vector2d

also, if possible, could you mark these as internal?